test/integration/puppetlabs/trapperkeeper/services/scheduler/scheduler_service_test.clj

```diff
@@ -62,37 +73,185 @@
+              (let [group-id-job-1 (guaranteed-start-interval-job service interval group-id)]
+                (is (= 4 (count-jobs service)))
+                (is (= 2 (count-jobs service group-id)))
+                (let [job-2 (guaranteed-start-interval-job service interval)]
+                  (is (= 5 (count-jobs service)))
+                  (is (= 2 (count-jobs service group-id)))
+                  ; stop the first job and wait a bit
+                  (stop-job service job-0)
+                  (Thread/sleep 100)
+                  (is (= 4 (count-jobs service)))
+                  (is (= 2 (count-jobs service group-id)))
+                  (stop-job service group-id-job-0)
+                  (Thread/sleep 100)
+                  (is (= 3 (count-jobs service)))
+                  (is (= 1 (count-jobs service group-id)))
+                  (stop-job service job-1)
+                  (Thread/sleep 100)
+                  (is (= 2 (count-jobs service)))
+                  (is (= 1 (count-jobs service group-id)))
+                  (stop-job service group-id-job-1)
+                  (Thread/sleep 100)
+                  (is (= 1 (count-jobs service)))
+                  (is (= 0 (count-jobs service group-id)))
+                  (stop-job service job-2)
+                  (Thread/sleep 100)
+                  (is (= 0 (count-jobs service)))
+                  (is (= 0 (count-jobs service group-id))))))))))
+
+  (testing "after reduces count when complete"
+    (with-app-with-empty-config app [scheduler-service]
+      (let [service (tk/get-service app :SchedulerService)
+            delay 100
+            wait-for-start (promise)
+            completed (promise)
+            job (fn []
+                  (deref wait-for-start)
+                  (deliver completed (System/currentTimeMillis)))]
+        (after service delay job)
+        (is (= 1 (count-jobs service)))
+        (deliver wait-for-start true)
+        (deref completed)
+        (Thread/sleep 100)
+        (is (= 0 (count-jobs service)))))))
+
+(deftest ^:integration test-stop-grouped-jobs
+  (testing "stop-grouped-jobs stops the jobs for a service"
     (with-app-with-empty-config app [scheduler-service]
       (let [service (tk/get-service app :SchedulerService)
             started (promise)
             stopped (promise)
-            start-time (atom 0)
-            completed (promise)
             job (fn []
-                  (reset! start-time (System/currentTimeMillis))
-                  (deliver started nil)
-                  (deref stopped)
-                  (deliver completed nil))
-            job-id (interspaced service 10 job)]
-        ; wait for the job to start
+                  (deliver started nil))
+            interval 10
+            group-id-0 :unique-group-id
+            group-id-1 :more-unique-group-id
+            ; create one job without a service id and two with one group-id
+            ; and a third with a different service id
```

Comment:
I think you mean `group id`, not `service id`.  This applies to a few places below as well.